ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -1311,6 +1341,8 @@ public boolean alreadyExists(PersistentIndex idx, FileSpan searchSpan, StoreKey
       copyCandidates.removeIf(
           copyCandidate -> isDuplicate(copyCandidate, duplicateSearchSpan, indexSegment.getStartOffset(),
               checkAlreadyCopied));
+      copyCandidates.removeIf(
+          copyCandidate -> isDeleteRequested(copyCandidate, selectedContainers));
```

Comment:
same here

Reply:
The logic here is to remove blob that belongs to containers being deleted. I would suggest considering edge case where the obsolete container has been removed from Helix account metadata ZNode, however, blobs from that container haven't been cleaned up on certain node (perhaps this node is down for a long time).

Reply:
Merged the duplicate code. And currently we've decided not to remove obsolete container metadata from Zookeeper. But will keep in mind if we change our decision.